storage/replicate_queue.go

```diff
@@ -83,12 +90,21 @@ func (rq replicateQueue) shouldQueue(now proto.Timestamp, repl *Replica) (should
 }
 
 func (rq replicateQueue) process(now proto.Timestamp, repl *Replica) error {
-	zone, err := lookupZoneConfig(rq.gossip, repl)
+	// Load the system config.
+	cfg, err := rq.gossip.GetSystemConfig()
 	if err != nil {
 		return err
 	}
 
+	// TODO(marc): shouldn't we be checking whether the range needs to be split?
```

Comment:
ok. I'll audit those and see if I can consolidate it. We're doing the same check so many times, that it would be worth begin smarter about it. It would also ensure that we don't miss said check.
Filed #2499.